docs/import-images.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Import images"
+---
+
+There are two ways to get images onto your nodes so that Pods can utilize them: pulling them on-demand or pre-importing them.
+
+
+## On-demand image pulling
+
+Kubernetes, by default, automatically pulls images when a Pod requires them if the image is not already present on the node. This behavior can be changed by using the [image pull policy](https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy) field of the Pod. When using the default `IfNotPresent` policy, containerd will pull the image from either upstream (default) or your [private registry](installation/private-registry.md) and store it in its image store. Users do not need to apply any additional configuration for on-demand image pulling to work.
+
+
+## Pre-import images
+:::info Version Gate
+The pre-importing of images while K3s is running feature is available as of January 2025 releases: v1.32.0+k3s1, v1.31.5+k3s1, v1.30.9+k3s1, v1.29.13+k3s1. Before that, K3s pre-imported the images only when booting.
+:::
+
+Pre-importing images onto the node is essential if you configure Kubernetes' `imagePullPolicy` as `Never`, for example, for security reasons. Another reason to do this is to reduce the time it takes for your K3s nodes to spin up.
+
+K3s includes two mechanisms to pre-import images into the containerd image store:
+
+<Tabs groupId = "import images" queryString>
+<TabItem value="Online image importing" default>
+
+Users can trigger the pulling of images into the containerd image store by adding a `.txt` file containing the image paths to the `/var/lib/rancher/k3s/agent/images` directory while K3s is running. A K3s controller will detect this file and trigger the image pull from an upstream (default) or [private registry](installation/private-registry.md).
```

Comment:
```suggestion
Users can trigger a pull of images into the containerd image store by placing a `.txt` file containing the image names, one per line, in the `/var/lib/rancher/k3s/agent/images` directory. The text file can be placed before K3s is started, or created/modified while K3s is running. K3s will sequentially pull the images via the CRI API, respecting the using the configuration from [`registries.yaml`](installation/private-registry.md).
```

Reply:
Thanks, I slightly modified the last sentence